bugsnag-plugin-android-exitinfo/src/main/java/com/bugsnag/android/TombstoneParser.kt

```diff
@@ -28,11 +30,42 @@ internal class TombstoneParser(
             if (listOpenFds) {
                 extractTombstoneFd(trace.openFdsList, fileDescriptorConsumer)
             }
+
+            if (includeLogcat) {
+                extractTombstoneLogBuffers(trace.logBuffersList, logcatConsumer)
+            }
         } catch (ex: Throwable) {
             logger.w("Tombstone input stream threw an Exception", ex)
         }
     }
 
+    private fun extractTombstoneLogBuffers(
+        logBuffersList: List<TombstoneProtos.LogBuffer>,
+        logcatConsumer: (String) -> Unit
+    ) {
+        val newLogList = mutableListOf<String>()
+        var priorityType: String
+        logBuffersList.forEach { logs ->
+            logs.logsList.forEach {
+                priorityType = when (it.priority) {
+                    2 -> "VERBOSE"
+                    3 -> "DEBUG"
+                    4 -> "INFO"
+                    5 -> "WARN"
+                    6 -> "ERROR"
+                    7 -> "ASSERT"
+                    else -> it.priority.toString()
+                }
+
+                newLogList.add(
+                    0,
+                    "\n${it.timestamp} ${it.tid} ${it.tag} $priorityType ${it.message}"
+                )
```

Comment:
This `add` is very expensive as it causes the entire array to be "moved" each time it's called.

In fact the entire use of a `List` is a bit heavy-weight. A better solution is probably to use a `StringBuilder`, either directly or through `buildString` and then read the lists backwards.